src/core/public/chrome/ui/header/header.scss

```diff
@@ -5,24 +5,70 @@
 
 .newTopNavHeader {
   z-index: 1000;
-  padding-left: $euiSizeXS;
+  padding: 0 $euiSizeL $euiSizeS;
   box-shadow: none;
   border-bottom: none;
   background: none;
+  height: auto;
+  gap: $euiSizeS;
 
   &.primaryHeader {
-    height: auto;
-    padding-top: $euiSizeM;
+    margin-top: $euiSizeM;
   }
 
   &.primaryApplicationHeader {
-    height: auto;
     padding-top: $euiSizeM - $euiSizeXS;
+    border-bottom: 1px solid $euiColorLightShade;
+
+    &:last-child {
+      margin-bottom: 0;
+    }
+  }
+
+  .headerAppActionMenu {
+    // stylelint-disable-next-line @osd/stylelint/no_modifying_global_selectors
+    & > .euiFlexGroup {
+      gap: $euiSizeS;
+    }
+
+    // stylelint-disable-next-line @osd/stylelint/no_modifying_global_selectors
+    & > .euiHeaderLinks > .euiHeaderLinks__list {
+      gap: $euiSizeS;
+
+      & > * {
+        margin: 0;
+      }
+    }
   }
 
   &:has(.headerDescriptionControl, .headerBottomControl) {
     height: auto;
   }
+
+  // stylelint-disable-next-line @osd/stylelint/no_modifying_global_selectors
+  & > .euiHeaderSection {
+    gap: $euiSizeS;
+
+    // stylelint-disable-next-line @osd/stylelint/no_modifying_global_selectors
+    & > .euiHeaderSectionItem:empty {
+      display: none;
+    }
+  }
+
+  // stylelint-disable-next-line @osd/stylelint/no_modifying_global_selectors
+  &:not(:has(>:not(:empty))),
+  .euiHeaderSectionItem:not(:has(>:not(:empty))) {
+    display: none;
+  }
+}
+
+.headerGlobalNav:has(.newTopNavHeader:not(.primaryApplicationHeader)) {
+  margin-bottom: -1 * $euiSizeS;
+}
+
+.newTopNavHeaderTitle {
+  line-height: 1;
+  font-size: 2rem;
```

Comment:
what is this for? Why aren't we using a token?

Reply:
This is a specific ask from UX.